core/domain/stats_services_test.py

```diff
@@ -466,6 +487,11 @@ def test_record_answers_exceeding_one_shard(self):
             self.assertEqual(
                 len(state_answers.get_submitted_answer_dict_list()), 600)
 
+        sample_answers = stats_services.get_sample_answers(
+            self.EXP_ID, self.exploration.version,
+            self.exploration.init_state_name)
+        self.assertLess(len(sample_answers), 600)
```

Comment:
Shouldn't this be 100? We ought to have tests verifying this properly handles the following cases:

- Main shard > 100 answers: returns 100 answers
- Main shard = 100 answers: returns 100 answers
- Main shard < 100 answers && shard count = 1 (only main shard): returns all answers of main shard
- Main shard < 100 answers && shard count > 1 (more than main shard): returns only all answers of main shard and no others